agent/apps/ai_guardrails/app_code/serializers.py

```
@@ -0,0 +1,93 @@
+    سریالایزر قانون رد-فلگ
+    """
+
+    class Meta:
+        model = RedFlagRule
+        fields = [
+            'id', 'name', 'pattern_type', 'pattern', 'category', 'severity',
+            'is_active', 'metadata', 'created_at', 'updated_at', 'created_by'
+        ]
+        read_only_fields = ['id', 'created_at', 'updated_at', 'created_by']
+
+    def validate(self, attrs):
```

Ruff (0.12.2)

Missing return type annotation for public function `validate`

(ANN201)